rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -117,10 +102,16 @@ public TopologyHandler getHandler(TopologyDescriptor topologyDescriptor) {
             !PAResourceManagerProperties.RM_TOPOLOGY_ENABLED.getValueAsBoolean()) {
             throw new TopologyDisabledException("Topology is disabled");
         }
+        if (topologyDescriptor instanceof BestProximityDescriptor || topologyDescriptor instanceof ThresholdProximityDescriptor) {
```

Comment:
No one likes it :-)
I thought precisely on the kind of implementation you suggest, but this change is targeted to 6.3.1 maintenance release. Adding a new property in TopologyDescriptor would imply an API change and I want to avoid that. I can implement right after this pull request something cleaner.